### ISSUE TITLE
Authorization expiration(?)

### ISSUE THREAD

**Issue by tridral** [astrada/google-drive-ocamlfuse#882]
From time to time (approximately weekly) my drives fail to mount.  I have to remove them, recreate them and reauthorize them, and then they're fine for a while.

The expiration may depend on time or usage  (I'm not sure).

I'd like to be able to diagnose the problem and if possible to set these drives so that they don't expire.

Has anyone else seen this issue? Do you have any ideas or solutions?

Many thanks,

'ö-Dzin

**Comment by astrada** [comment]
Yes, sorry, if you set your app in the OAuth consent screen to Testing, the validity of refresh tokens is limited to one week. I don't think there is a workaround.

**Comment by tridral** [comment]
Dear Alessandro,

Thank you very much for your reply.  This explains the behaviour and gives me the option of changing how things work.

best wishes,

'ö-Dzin

**Comment by astrada** [comment]
It looks like you can use production mode without Google verification. See rclone/rclone#7963

**Comment by bobhy** [comment]
I just encountered the expiration for the first time.  When I went to move the credential into Production, as recommended above, Google provided this confirmation box:
>>> Push to production?
Your app will be available to any user with a Google Account.
>>>
>>>You do not need to submit your app for verification. If you change your app's configuration in the future, like adding more than 10 domains, uploading a logo, or requesting sensitive or restricted scopes, you will need to [submit for verification ](https://support.google.com/cloud/answer/9110914).

... Update ... this seems to have worked.  I couldn't access the mountpoint until I forced reauthentication via browser (and Google gave me the 'this app has not been verified by Google', but allowed to to proceed if I was willing to be unsafe).  I deleted the whole `~/.gdfuse/default` folder and recreated it via `google-drive-ocamlfuse -id ...` in order to get the browser auth.  Maybe there's a less destructive way?  In any case, I won't know for another 7 days whether it's really being more patient...

...

**Comment by DirkNie** [comment]
Hi, is there a solution available?
I have done the above way by deleting the ~/.gfuse/default folder, but I get the Google Access_denied 
![grafik](https://github.com/user-attachments/assets/13382cbc-d449-472c-a6d9-10dacf0334d4)
And there is actually no access for 7 days available anymore. 

Is there a way to become an developer tester?
TIA Dirk

**Comment by zer0shock** [comment]
Same issue here - 
![Untitled](https://github.com/user-attachments/assets/1912228d-833b-498b-a704-a09eb45dbc41)

**Comment by superuser-does** [comment]
You can re-authorise by deleting the token held in `~/.cache/state`. It would be great if the software supported the option to auto-expire tokens every 7 days (or something more smart than that if possible), @astrada.
